apps/root/src/services/conctactListService.ts

```diff
@@ -14,83 +21,104 @@ export default class ContactListService implements IContactListService {
 
   meanApiService: MeanApiService;
 
-  contactList: ContactList = [];
+  walletService: WalletService;
+
+  labelService: LabelService;
 
   constructor(
     accountService: AccountService,
     providerService: ProviderService,
     meanApiService: MeanApiService,
-    contractService: ContractService
+    contractService: ContractService,
+    walletService: WalletService,
+    labelService: LabelService
   ) {
+    super({ contactList: [] });
     this.accountService = accountService;
     this.providerService = providerService;
     this.contractService = contractService;
     this.meanApiService = meanApiService;
+    this.walletService = walletService;
+    this.labelService = labelService;
   }
 
-  async addContact(contact: Contact): Promise<void> {
+  get contactList() {
+    return this.serviceData.contactList;
+  }
+
+  set contactList(contactList) {
+    this.serviceData = { ...this.serviceData, contactList };
```

Comment:
Once we get to positionService we are going to need to figure out a better way to handle this, since the setter is not going to active when changing subproperties of the object

Reply:
I already faced that issue while testing this 😀. Updating the whole object works here, but it might have significant downsides in the positionService